lib/components/form/date-time-button.tsx

```diff
@@ -0,0 +1,190 @@
+  arrow,
+  FloatingFocusManager,
+  offset,
+  safePolygon,
+  shift,
+  useClick,
+  useDismiss,
+  useFloating,
+  useHover,
+  useInteractions,
+  useRole
+} from '@floating-ui/react'
+import { FormattedMessage } from 'react-intl'
+import React, { HTMLAttributes, useCallback, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import InvisibleA11yLabel from '../util/invisible-a11y-label'
+
+import { activeCss, boxShadowCss, buttonPixels } from './batch-styled'
+import DateTimeModal from './date-time-modal'
+import DateTimePreview from './date-time-preview'
+
+const ButtonWrapper = styled.span`
+  position: relative;
+
+  & > button {
+    border-radius: 5px;
+    border: none;
+    cursor: pointer;
+    font-size: 12px;
+    height: ${buttonPixels}px;
+    padding: 4px 5px;
+    text-align: left;
+    transition: all 250ms cubic-bezier(0.27, 0.01, 0.38, 1.06);
+    white-space: nowrap;
+    width: 120px;
+  }
+
+  & > button:hover,
+  & > button[aria-expanded='true'] {
+    ${activeCss}
+    ${boxShadowCss}
+  }
+`
+
+const HoverPanel = styled.div`
+  min-width: min(400px, 100vw);
+  max-width: 100vw;
+  padding: 0 10px;
+  width: 33vw;
+  z-index: 100;
+`
+
+// Ideally, import styles below from mode selector package.
+const HoverInnerContainer = styled.div`
+  background: #fff;
+  border-radius: 4px;
+  padding: 0px 20px 10px;
+  ${boxShadowCss}
+`
+
+const Arrow = styled.div`
+  background: #fff;
+  height: 10px;
+  margin-top: -5px;
+  position: absolute;
+  transform: rotate(-45deg);
+  width: 10px;
+  ${boxShadowCss}
+`
+
+interface DateTimeButtonProps extends HTMLAttributes<HTMLSpanElement> {
+  open: boolean
+  setOpen: (arg: boolean) => void
+}
+
+/**
+ * Button and popup component for the date & time selector.
+ */
+export default function DateTimeButton({
+  open,
+  setOpen,
+  style
+}: DateTimeButtonProps): JSX.Element {
+  const arrowRef = useRef(null)
+  // State used to keep the popup open when triggered by keyboard while moving the mouse around.
+  const [openWithKeyboard, setOpenWithKeyboard] = useState(false)
+  const onOpenChange = useCallback(
+    (value) => {
+      setOpen(value)
+      if (!value) {
+        setOpenWithKeyboard(false)
+      }
+    },
+    [setOpen, setOpenWithKeyboard]
+  )
+  const {
+    context,
+    floating,
+    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
+    reference,
+    strategy,
+    x,
+    y
+  } = useFloating({
+    middleware: [offset(8), shift(), arrow({ element: arrowRef })],
+    onOpenChange,
+    open
+  })
+
+  const { getFloatingProps, getReferenceProps } = useInteractions([
+    useHover(context, {
+      // Enable hover only if no popup has been triggered via keyboard.
+      // (This keeps the keyboard-triggered popups open while moving mouse around.)
+      enabled: !openWithKeyboard,
+      handleClose: safePolygon({
+        blockPointerEvents: false,
+        buffer: 0,
+        restMs: 500
+      })
+    }),
+    useClick(context),
```

Comment:
I'm not sure I agree with a click opening the popup, as this doesn't match the mode selector behavior

Reply:
> I'm not sure I agree with a click opening the popup, as this doesn't match the mode selector behavior

What do you mean by that? Hover is already enabled. Click is still needed for keyboard access.

Reply:
The hover is good, but having the popup stick on click is behavior I think we should disable

Reply:
Thanks for clarifying, I figured out that fix in https://github.com/opentripplanner/otp-react-redux/pull/955/commits/97a0eb06172395b7cfb7113a081703c32dc4d380.